specification/compute/resource-manager/Microsoft.Compute/stable/2020-09-30/disk.json

```diff
@@ -0,0 +1,3303 @@
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/DiskRestorePoint"
+          },
+          "description": "A list of disk restore points."
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "The uri to fetch the next page of disk restore points. Call ListNext() with this to fetch the next page of disk restore points."
+        }
+      },
+      "required": [
+        "value"
+      ],
+      "description": "The List Disk Restore Points operation response."
+    },
+    "DiskRestorePointProperties": {
+      "properties": {
+        "timeCreated": {
+          "type": "string",
+          "format": "date-time",
+          "description": "The timestamp of restorePoint creation",
+          "readOnly": true
+        },
+        "sourceResourceId": {
+          "type": "string",
+          "description": "arm id of source disk",
+          "readOnly": true
+        },
+        "osType": {
+          "type": "string",
+          "readOnly": true,
+          "description": "The Operating System type.",
+          "enum": [
+            "Windows",
+            "Linux"
+          ],
+          "x-ms-enum": {
+            "name": "OperatingSystemTypes",
+            "modelAsString": false
+          }
+        },
+        "hyperVGeneration": {
+          "type": "string",
+          "description": "The hypervisor generation of the Virtual Machine. Applicable to OS disks only.",
+          "enum": [
+            "V1",
+            "V2"
+          ],
+          "x-ms-enum": {
+            "name": "HyperVGeneration",
+            "modelAsString": true
+          }
+        },
+        "purchasePlan": {
+          "$ref": "#/definitions/PurchasePlan",
+          "description": "Purchase plan information for the the image from which the OS disk was created."
+        },
+        "familyId": {
+          "type": "string",
+          "description": "id of the backing snapshot's MIS family",
+          "readOnly": true
+        },
+        "sourceUniqueId": {
+          "type": "string",
+          "description": "unique incarnation id of the source disk",
+          "readOnly": true
+        },
+        "encryption": {
+          "$ref": "#/definitions/Encryption",
+          "description": "Encryption property can be used to encrypt data at rest with customer managed keys or platform managed keys.",
+          "readOnly": true
+        }
+      },
+      "description": "Properties of an incremental disk restore point"
+    },
+    "PrivateLinkResourceListResult": {
+      "properties": {
+        "value": {
+          "type": "array",
+          "description": "Array of private link resources",
+          "items": {
+            "$ref": "#/definitions/PrivateLinkResource"
+          }
+        }
+      },
+      "description": "A list of private link resources"
+    },
+    "PrivateEndpointConnectionListResult": {
```

Comment:
You should consider making this schema pageable by annotating it with x-ms-pageable and adding a nextLink property. That way, even if you don't implement it now, you can implement it later without a breaking change.

Reply:
Good point. Added pageable in line 1553 for list PrivateEndpointConnection and in line 1649 for list DiskRestorePoints